src/System.Text.Primitives/System/Text/EncodingData.cs

```diff
@@ -365,8 +345,153 @@ private bool VerifySuffix(ReadOnlySpan<byte> buffer, int codeUnitIndex, Symbol s
         // CC = the max value, and DD = the max value's index in the same coord-system as BB.
         struct ParsingTrieNode
         {
-            public byte valueOrNumChildren;
+            public byte ValueOrNumChildren;
             public int IndexOrSymbol;
         }
+        
```

Comment:
Also, could you describe the algorithm and the role of each of the types in the comments.